actions/reservation.py

```diff
@@ -27,3 +37,87 @@ def get_actual_after_buffer(self) -> datetime.timedelta:
             after = reservation_unit.actions.get_actual_after_buffer(self.reservation.end)
             buffer_time_after = max(after, buffer_time_after)
         return buffer_time_after
+
+    def to_ical(self, *, site_name: str) -> bytes:
+        language: Lang = (  # type: ignore[assignment]
+            self.reservation.reservee_language or self.reservation.user.get_preferred_language()
+        )
+
+        ical_event = Event()
+        # This should be unique such that if another iCal file is created
+        # for the same reservation, it will be the same as the previous one.
+        uid = f"varaamo.reservation.{self.reservation.pk}@{site_name}"
+        summary = self.get_ical_summary(language=language)
+        description = self.get_ical_description(language=language)
+
+        ical_event.add(name=EventProperty.UID, value=uid)
+        ical_event.add(name=EventProperty.DTSTAMP, value=local_datetime())
+        ical_event.add(name=EventProperty.DTSTART, value=self.reservation.begin.astimezone(DEFAULT_TIMEZONE))
+        ical_event.add(name=EventProperty.DTEND, value=self.reservation.end.astimezone(DEFAULT_TIMEZONE))
+
+        ical_event.add(name=EventProperty.SUMMARY, value=summary)
+        ical_event.add(name=EventProperty.DESCRIPTION, value=description, parameters={"FMTTYPE": "text/html"})
+        ical_event.add(name=EventProperty.LOCATION, value=self.get_location_string())
```

Comment:
Not in the ticket spec but could also include latitude and longitude of the unit in `Event.GEO`: https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.1.6

Reply:
Added.